test/OpenTelemetry.Instrumentation.AspNetCore.Tests/RouteTests/RoutingTests.cs

```diff
@@ -30,77 +28,133 @@ public RoutingTests(RoutingTestFixture fixture)
         var metricPoint = Assert.Single(metricPoints);
 
-        GetTagsFromActivity(activity, out var activityHttpStatusCode, out var activityHttpMethod, out var activityHttpRoute);
         GetTagsFromMetricPoint(metricPoint, out var metricHttpStatusCode, out var metricHttpMethod, out var metricHttpRoute);
 
-        Assert.Equal(testCase.ExpectedStatusCode, activityHttpStatusCode);
         Assert.Equal(testCase.ExpectedStatusCode, metricHttpStatusCode);
-        Assert.Equal(testCase.HttpMethod, activityHttpMethod);
         Assert.Equal(testCase.HttpMethod, metricHttpMethod);
 
         // TODO: The CurrentHttpRoute property will go away. They only serve to capture status quo.
         // If CurrentHttpRoute is null, then that means we already conform to the correct behavior.
         var expectedHttpRoute = testCase.CurrentHttpRoute ?? testCase.ExpectedHttpRoute;
-        Assert.Equal(expectedHttpRoute, activityHttpRoute);
-
         var expectedMetricRoute = testCase.ExpectedMetricRoute ?? expectedHttpRoute;
         Assert.Equal(expectedMetricRoute, metricHttpRoute);
 
-        // Activity.DisplayName should be a combination of http.method + http.route attributes, see:
-        // https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#name
-        var expectedActivityDisplayName = string.IsNullOrEmpty(expectedHttpRoute)
-            ? testCase.HttpMethod
-            : $"{testCase.HttpMethod} {expectedHttpRoute}";
-
-        Assert.Equal(expectedActivityDisplayName, activity.DisplayName);
-
-        var testResult = new RoutingTestResult
+        var testResult = new MetricRoutingTestResult(testCase)
         {
             IdealHttpRoute = testCase.ExpectedHttpRoute,
-            ActivityDisplayName = activity.DisplayName,
-            ActivityHttpRoute = activityHttpRoute,
             MetricHttpRoute = metricHttpRoute,
-            TestCase = testCase,
             RouteInfo = RouteInfo.Current,
         };
 
-        this.fixture.AddTestResult(testResult);
+        this.fixture.AddMetricsTestResult(testResult);
+    }
+
+    private static async Task<bool> TryWaitUntilAny<T>(ICollection<T> collection, Action reflush, Func<T, bool>? filter = null)
+    {
+        // flush and see if data is immediately available
+        reflush();
+
+        for (var i = 0; i < 10; i++)
+        {
+            if (collection.Count > 0)
+            {
+                // nothing to filter
+                if (filter == null)
+                {
+                    return true;
+                }
+
+                foreach (var item in collection)
+                {
+                    // filter is defined and succeeds fiding elements
+                    if (filter(item))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            reflush();
+
+            // Add time to process the flush
+            await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
Maybe make this 1.5s? I've fixed a few flaky tests recently where CI was _slightly_ too slow and they'd fail.

Alternatively can you just poll for the expected result (with a maximum time to wait), rather than just unconditionally wait then assert?